tracpro/polls/utils.py

```diff
@@ -49,3 +49,9 @@ def extract_words(text, language):
     words = re.split(r"[^\w'-]", text.lower(), flags=re.UNICODE)
     ignore_words = ignore_words
     return [w for w in words if w not in ignore_words and len(w) > 1]
+
+
+def natural_sort_key(text):
+    """Key to sort text in a humanized way, e.g., 11 should come before 100."""
+    return [int(t) if t.isdigit() else t.lower()
+            for t in re.split('([0-9]+)', text)]
```

Comment:
I was pretty sure the previous version took care of the ["10-100", "10-11"] example, but maybe it didn't. That's fine. I still find this somewhat unreadable. Any reason not to split it out into multiple lines for clarity with a separate return statement?